proxy/src/main/java/com/wavefront/agent/listeners/tracing/JaegerPortUnificationHandler.java

```diff
@@ -0,0 +1,218 @@
+  private final Supplier<ReportableEntityPreprocessor> preprocessorSupplier;
+  private final Sampler sampler;
+  private final boolean alwaysSampleErrors;
+  private final String proxyLevelApplicationName;
+  private final Set<String> traceDerivedCustomTagKeys;
+
+  private final Counter discardedTraces;
+  private final Counter discardedBatches;
+  private final Counter processedBatches;
+  private final Counter failedBatches;
+  private final Counter discardedSpansBySampler;
+  private final ConcurrentMap<HeartbeatMetricKey, Boolean> discoveredHeartbeatMetrics;
+  private final ScheduledExecutorService scheduledExecutorService;
+
+  private final static String JAEGER_VALID_PATH = "/api/traces/";
+  private final static String JAEGER_VALID_HTTP_METHOD = "POST";
+
+  @SuppressWarnings("unchecked")
+  public JaegerPortUnificationHandler(String handle,
+                                      final HealthCheckManager healthCheckManager,
+                                      ReportableEntityHandlerFactory handlerFactory,
+                                      @Nullable WavefrontSender wfSender,
+                                      Supplier<Boolean> traceDisabled,
+                                      Supplier<Boolean> spanLogsDisabled,
+                                      @Nullable Supplier<ReportableEntityPreprocessor> preprocessor,
+                                      Sampler sampler,
+                                      boolean alwaysSampleErrors,
+                                      @Nullable String traceJaegerApplicationName,
+                                      Set<String> traceDerivedCustomTagKeys) {
+    this(handle, healthCheckManager,
+        handlerFactory.getHandler(HandlerKey.of(ReportableEntityType.TRACE, handle)),
+        handlerFactory.getHandler(HandlerKey.of(ReportableEntityType.TRACE_SPAN_LOGS, handle)),
+        wfSender, traceDisabled, spanLogsDisabled, preprocessor, sampler, alwaysSampleErrors,
+        traceJaegerApplicationName, traceDerivedCustomTagKeys);
+  }
+
+  @VisibleForTesting
+  JaegerPortUnificationHandler(String handle,
+                               final HealthCheckManager healthCheckManager,
+                               ReportableEntityHandler<Span> spanHandler,
+                               ReportableEntityHandler<SpanLogs> spanLogsHandler,
+                               @Nullable WavefrontSender wfSender,
+                               Supplier<Boolean> traceDisabled,
+                               Supplier<Boolean> spanLogsDisabled,
+                               @Nullable Supplier<ReportableEntityPreprocessor> preprocessor,
+                               Sampler sampler,
+                               boolean alwaysSampleErrors,
+                               @Nullable String traceJaegerApplicationName,
+                               Set<String> traceDerivedCustomTagKeys) {
+    super(TokenAuthenticatorBuilder.create().build(), healthCheckManager, handle);
```

Comment:
Since this is http, it's really easy to add true auth support to this port as well, just pass the TokenAuthenticator as an argument from PushAgent.

Reply:
updated to pass in the TokenAuthenticator